src/validation/validationSchema.ts

```diff
@@ -0,0 +1,170 @@
+ */
+
+export default {
+    Common: {
+        Next: true,
+        End: true,
+        Comment: true,
+        InputPath: true,
+        OutputPath: true,
+        Type: true
+    },
+    StateTypes: {
+        Pass: {
+            Properties: {
+                Result: true,
+                ResultPath: true,
+                Parameters: true
+            },
+            hasCommonProperties: true
+        },
+        Task: {
+            Properties: {
+                Resource: true,
+                Parameters: true,
+                ResultPath: true,
+                TimeoutSeconds: true,
+                HeartbeatSeconds: true,
+                Retry: {
+                    'Fn:ArrayOf': 'Retrier'
+                },
+                Catch: {
+                    'Fn:ArrayOf': 'Catcher'
+                }
+            },
+            hasCommonProperties: true
+        },
+        Choice: {
+            Properties: {
+                Comment: true,
+                InputPath: true,
+                OutputPath: true,
+                Type: true,
+                Choices: {
+                    'Fn:ArrayOf': 'ChoiceRule'
+                },
+                Default: true,
+            }
+        },
+        Wait: {
+            Properties: {
+                Seconds: true,
+                Timestamp: true,
+                SecondsPath: true,
+                TimestampPath: true
+            },
+            hasCommonProperties: true
+        },
+        Succeed: {
+            Properties: {
+                Type: true,
+                Comment: true,
+                InputPath: true,
+                OutputPath: true,
+            }
+        },
+        Fail: {
+            Properties: {
+                Cause: true,
+                Error: true,
+                Comment: true,
+                Type:  true
+            }
+        },
+        Parallel: {
+            Properties: {
+                Branches: true,
+                ResultPath: true,
+                Parameters: true,
+                Retry: {
+                    'Fn:ArrayOf': 'Retrier'
+                },
+                Catch: {
+                    'Fn:ArrayOf': 'Catcher'
+                }
+            },
+            hasCommonProperties: true
+        },
+        Map: {
+            Properties: {
+                Iterator: true,
+                ItemsPath: true,
+                MaxConcurrency: true,
+                ResultPath: true,
+                Parameters: true,
+                Retry: {
+                    'Fn:ArrayOf': 'Retrier'
+                },
+                Catch: {
+                    'Fn:ArrayOf': 'Catcher'
+                }
+            },
+            hasCommonProperties: true
+        }
+    },
+    ReferenceTypes: {
+        ComparisonOperators: {
+            And: {
+                'Fn:ArrayOf': 'NestedChoiceRule'
+            },
+            BooleanEquals: true,
+            Not: {
+                'Fn:ValueOf': 'ChoiceRule'
```

Comment:
This is also a `NestedChoiceRule`
```suggestion
                'Fn:ValueOf': 'NestedChoiceRule'
```

Reply:
changed